CONVENTIONS.md

```diff
@@ -0,0 +1,78 @@
+# Agreed Conventions
+
+## Code structure
+
+To enable ease of understanding and consistency, we recommend following the structure as such:
+
+> Note: Add headers to the respective sections
+
+```tsx
+// component-file.tsx
+
+// Import statements
+
+/**
+ * For local props. If the props are to be exported
+ * add them to types.ts
+ */
+interface Props {
+    a: string;
+}
+
+export const MyComponent = ({ a }: Props) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+
+    /**
+     * Name event handlers using `handle` prefix and the name of the action after.
+     * E.g. handleClick, handleChange
+     */
+
+    const handleClick = (event: React.KeyboardEvent<HTMLButtonElement>) => {
+        // do something...
+    };
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+
+    /**
+     * If the helper function can be extracted (i.e. doesnt rely on any internal
+     * constant/state value), then consider putting it outside of the component
+     * instead
+     */
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    return <InnerComponent />;
+};
+```
+
+<br />
+
+## Prop specification
+
+Here are some guidelines on prop specification:
+
+-   For callbacks, denote with `on` and the action. E.g. `onClick`, `onDismiss`
+-   For optional props, use a Union type and add `undefined` to it
```

Comment:
would be good to include these ones (I tend to miss out on them also -_-)
- include `data-testid` as well as `className`, `id` if not extending from a standard HTML element
- prefer string literals in kebab-case